test/api/students_api_test.rb

```diff
@@ -0,0 +1,54 @@
+require 'test_helper'
+require 'user'
+
+class StudentsApiTest < ActiveSupport::TestCase
+  include Rack::Test::Methods
+  include TestHelpers::AuthHelper
+  include TestHelpers::JsonHelper
+
+  def app
+    Rails.application
+  end
+
+  def test_get_students_with_authorization
+    # Create unit
+    newUnit = FactoryBot.create(:unit)
+
+    # The get that we will be testing.
+    get with_auth_token "/api/students/?unit_id=#{newUnit.id}", newUnit.main_convenor_user
+
+    response_keys = %w(first_name last_name)
+
+    # check the response
+    last_response_body.each do | data |
+      pro = Project.find(data['project_id'])
```

Comment:
Get the students from the unit - otherwise they could be any students...

Reply:
It is fixed. Thank you.